sanic/request.py

```diff
@@ -304,19 +305,21 @@ def parse_multipart_form(body, boundary):
             if form_header_field == 'content-disposition':
                 if 'filename' in form_parameters:
                     file_name = form_parameters['filename']
-                field_name = form_parameters.get('name')
+                field_name = form_parameters['name']
```

Comment:
please leave this as `field_name = form_parameters.get('name')` to avoid throwing an error here and causing the sanic process to die

Reply:
Okay, good. The "name" parameter is required in the Content-Disposition header according to the specification, but it is indeed nice to not cause a crash when it is missing of course.

Reply:
It is now weird when it is missing though... There will be a 'None' key in the files/form dictionary.

Reply:
hmm is there a different default that makes sense? Just don't want the server to crash even if it's a rare edge case

Reply:
I think we should not allow the field/file to end up in the request at all if there is no 'name' parameter in the Content-Disposition header. Like I said, it is required by the spec.

We can't define a default value because you should be able to POST any field name in the form-data. For example, if we say the default value is the string value 'None', then the behavior will be weird when somebody wants to use 'None' as a field name and has a form-field without a 'name'. It's a big edge case of course, but still...

Reply:
I made a commit to this PR to omit the form field from the request when it does not have a name parameter. When it is omitted, there is a log sent to the logger.DEBUG. I wasn't sure what the logging policy is though, should it go to error_logger instead?